odoo/tools/lru.py

```diff
@@ -1,33 +1,21 @@
 # -*- coding: utf-8 -*-
-# taken from http://code.activestate.com/recipes/252524-length-limited-o1-lru-cache-implementation/
+import collections
 import threading
 
 from .func import synchronized
 
 __all__ = ['LRU']
 
-class LRUNode(object):
-    __slots__ = ['prev', 'next', 'me']
-    def __init__(self, prev, me):
-        self.prev = prev
-        self.me = me
-        self.next = None
-
 class LRU(object):
     """
-    Implementation of a length-limited O(1) LRU queue.
-    Built for and used by PyPE:
-    http://pype.sourceforge.net
-    Copyright 2003 Josiah Carlson.
+    Implementation of a length-limited O(1) LRU map.
+
+    Original Copyright 2003 Josiah Carlson, later rebuilt on OrderedDict.
     """
-    def __init__(self, count, pairs=[]):
+    def __init__(self, count, pairs=()):
         self._lock = threading.RLock()
         self.count = max(count, 1)
-        self.d = {}
-        self.first = None
-        self.last = None
-        for key, value in pairs:
-            self[key] = value
+        self.d = collections.OrderedDict(pairs)
```

Comment:
change to fix: previously initialising the cache with more entries than the specified size / count would trim the entry set down